pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/forward/ForwardIndexReaderFactory.java

```diff
@@ -41,7 +41,15 @@
 
 
 public class ForwardIndexReaderFactory extends IndexReaderFactory.Default<ForwardIndexConfig, ForwardIndexReader> {
-  public static final ForwardIndexReaderFactory INSTANCE = new ForwardIndexReaderFactory();
+  private static ForwardIndexReaderFactory _instance = new ForwardIndexReaderFactory();
+
+  public static void setInstance(ForwardIndexReaderFactory factory) {
```

Comment:
I think either the variable should be volatile or we should clearly indicate here that the instance can only be changed at startup.

Reply:
Ack will mark as volatile.